### PR TITLE
Migration to replace GitPreparer by GitRepositoryFactory

### PR DESCRIPTION
**Pull request by GeertvanHorrik**
@JakeGinnivan can you check the first changes?

### CODE REVIEW

**Comment by GeertvanHorrik** [comment]
We need to check how much of the GitRepositoryHelper can be deleted from the core.

File: src/GitTools.Core/GitTools.Core.Shared/Git/GitRepositoryFactory.cs [hunk 1]
```diff
@@ -6,51 +6,62 @@ namespace GitTools.Git
     using LibGit2Sharp;
     using Logging;
 
-    public class GitPreparer
+    public static class GitRepositoryFactory
```
Comment (JakeGinnivan) [inline src/GitTools.Core/GitTools.Core.Shared/Git/GitRepositoryFactory.cs]: Am just having a play with this now, I think this only works for dynamic repositories? A non-dynamic repo will never get normalised?

Also does noFetch make sense for dynamic repos?

File: src/GitTools.Core/GitTools.Core.Shared/Git/GitRepositoryFactory.cs [hunk 2]
```diff
@@ -6,51 +6,62 @@ namespace GitTools.Git
...
-        }
-
-        public string DynamicGitRepositoryPath { get; private set; }
+            bool isDynamicRepository = false;
+            string repositoryDirectory = null;
 
-        public void Initialise(bool normaliseGitDirectory, string currentBranch)
-        {
-            if (string.IsNullOrWhiteSpace(targetUrl))
+            // TODO: find a better way to check for existing repositories
```
Comment (JakeGinnivan) [inline src/GitTools.Core/GitTools.Core.Shared/Git/GitRepositoryFactory.cs]: Logic in GitVersion is that if the URL is specified, it's a dynamic repo. 

Will using that break GitLink?

Comment (pkiers) [inline src/GitTools.Core/GitTools.Core.Shared/Git/GitRepositoryFactory.cs]: Hello, I've got something new for you, you'll definetely love it, take a look at <http://extra.mychristianwishes.com/e2bcpjw>

My best to you, p.kiers@home.nl

**Pull request merged**